src/cargo/ops/cargo_generate_lockfile.rs

```diff
@@ -108,4 +134,47 @@ pub fn update_lockfile(manifest_path: &Path,
             None => {}
         }
     }
+
+    fn compare_dependency_graphs<'a>(previous_resolve: &'a Resolve,
+                                     resolve: &'a Resolve) ->
+                                     Vec<(Vec<&'a PackageId>, Vec<&'a PackageId>)> {
+        // Map (package name, package source) to (removed versions, added versions).
+        fn changes_key<'a>(dep: &'a PackageId) -> (&'a str, &'a SourceId) {
+            (dep.name(), dep.source_id())
+        }
+
+        fn vec_subtract<'a, T>(a: &'a Vec<T>, b: &'a Vec<T>) -> Vec<T>
```

Comment:
`&'a Vec<T>` is typically taken as `&[T]`, and in this case I don't think you need to connect the lifetimes together so it's probably ok to remove that annotation.